compiler/src/iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp

```diff
@@ -83,6 +83,12 @@ static llvm::cl::opt<bool> clDisableVectorPeeling(
                    "heuristics to select the strategy)."),
     llvm::cl::init(false));
 
+static llvm::cl::opt<bool> clEnableScalableVectorization(
+    "iree-llvmcpu-enable-scalable-vectorization",
+    llvm::cl::desc("Enable scalable vectorization if it is supported by the "
+                   "target (e.g., +sve, +sve2 and/or +sme feature flags)"),
+    llvm::cl::init(false));
```

Comment:
Not a blocker, but I would suggest initialising this to `true`. This way:
* `--iree-llvmcpu-target-cpu-features="+sve"` would imply scalable vectorisation (current behaviour),
* `--iree-llvmcpu-target-cpu-features="+sve" -iree-llvmcpu-enable-scalable-vectorization=false` would imply fixed-width vectorisation.

In other words, `+sve` would default to "scalable auto-vec" and users would have to pro-actively disable it. Not a strong opinion - this would just feel more natural. Somebody will end up requiring an extra flag -  I don't mind if that's me.

Reply:
I agree that that should be the ultimate goal but if we are enabling SVE for any flag internally implying the "+sve" feature (e.g., +v9a), we can't default to scalable vectorization just yet. Scalable vectorization has made a terrific progress but I think a bit more work is needed for the default enablement. It would require a bit more performance tuning to match the fixed-length path and we are also hitting compilation issues. If we enabled it by default, any modern Arm CPU (any Cortex-X3 (Pixel 8, new Snapdragons, new Mediatek, etc.) would hit these issues by default.

Would that be acceptable?

Reply:
> Would that be acceptable?

Totally, thanks for the discussion!